syntax = "proto3";

package panda_studio.panda_agent;
option go_package = "github.com/panda-re/panda_studio/panda_agent/pb";

service PandaAgent {
    rpc StartAgent (StartAgentRequest) returns (StartAgentResponse) {}
    rpc StopAgent (StopAgentRequest) returns (StopAgentResponse) {}
    rpc RunCommand (RunCommandRequest) returns (RunCommandReply) {}
}

message StartAgentRequest {}
message StartAgentResponse {}
message StopAgentRequest {}
message StopAgentResponse {}

message RunCommandRequest {
    string command = 1;
}

message RunCommandReply {
    int32 statusCode = 1;
    string output = 2;
}
